### PR TITLE
Feature/pcastell/nnr updates

### PR DESCRIPTION
**Pull request by patricia-nasa**
Updates to MODIS NNR code to work with python3 and handle Angstrom Exponent targets. 
Also other bug fixes found while testing this code.
- fix in trj_sampler for ICARTT files that have missing values in the location/time
- fix in py_ods that was trying to do element wise comparisons on tuples and lists
- updated GMAOpyobs to v1.0.2 so mxd04.writeods writes out the original AOD values when calculating the NNR AOD.

### CODE REVIEW

File: src/Applications/GAAS_App/mxd04_nnr.py [hunk 1]
```diff
@@ -304,6 +313,14 @@ def speciate(self,aer_x,Verbose=False):
         except:
             pass   # ignore it for systems without nitrates
 
+        # Handle brown carbon
+        # --------------------
+        try:
+            self.sampleFile(aer_x,onlyVars=('BRCEXTTAU',),Verbose=Verbose)
```
Comment (vbuchard) [inline src/Applications/GAAS_App/mxd04_nnr.py:319]: You may want to update to the latest name for BRC used in GOCART2G which is BREXTTAU.

**Review by vbuchard — APPROVED** [review]
It looks good to me. You may want to update the BRC name to the latest.

File: src/Applications/GAAS_App/mxd04_nnr.py (hunk 1 above)
Comment (patricia-nasa) [reply src/Applications/GAAS_App/mxd04_nnr.py]: thanks for catching that. 
Can you re-approve.

**Review by vbuchard — APPROVED** [review]

File: src/Applications/GAAS_App/mxd04_nnr.py [hunk 2]
```diff
@@ -304,6 +313,14 @@ def speciate(self,aer_x,Verbose=False):
         except:
             pass   # ignore it for systems without nitrates
 
+        # Handle brown carbon
+        # --------------------
+        try:
+            self.sampleFile(aer_x,onlyVars=('BREXTTAU',),Verbose=Verbose)
+            self.fcc += self.sample.BRCEXTTAU / s.TOTEXTTAU
```
Comment (vbuchard) [inline src/Applications/GAAS_App/mxd04_nnr.py:320]: You need to update this one as well.

**Review by vbuchard — APPROVED** [review]

**Pull request merged**